ehrapy/logger.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from logging import CRITICAL, ERROR, WARNING, INFO, DEBUG
+from datetime import datetime, timezone, timedelta
+from functools import partial, update_wrapper
+from typing import Optional
+
+HINT = (INFO + DEBUG) // 2
+logging.addLevelName(HINT, "HINT")
+
+
+class _RootLogger(logging.RootLogger):
+    def __init__(self, level):
+        super().__init__(level)
+        self.propagate = False
+        _RootLogger.manager = logging.Manager(self)
+
+    def log(
+        self,
+        level: int,
+        msg: str,
+        *,
+        extra: Optional[dict] = None,
+        time: datetime = None,
+        deep: Optional[str] = None,
+    ) -> datetime:
+        from ehrapy import settings
```

Comment:
We are using this in multiple places across the whole file; maybe we should just import this one once at the top?

Reply:
Yes, makes sense, will be done.